### DISCUSSION TITLE
I have a question about the barHeight when render the wave?

### DISCUSSION THREAD

**Discussion by Idletalk** [katspaugh/wavesurfer.js#3052] [Q&A]
The code is here https://github.com/katspaugh/wavesurfer.js/blob/main/src/renderer.ts#L287.  
I'm curious why the barHeight isn't the sum of the minimum and maximum values of topChannel[0], but the sum of the maximum value of Math.abs(topChannel[i]) and Math.abs(bottomChannel[i])

**Comment by katspaugh** [comment]
Because the top half of the waveform is the left channel and the bottom part is the right channel. It takes the maximum value per sample for both.

**Reply by Idletalk** [reply]
Thank you for your answer, I am new to this direction. Is there any theory behind this statement about “the top half of the waveform is the left channel and the bottom part is the right channel” or What articles can I know

**Reply by katspaugh** [reply]
User name checks out...
Feel free to read this article: https://best-actress-dd7.notion.site/What-is-a-waveform-2643c77db21d4fa8bf11fbd3f29667db

**Reply by Idletalk** [reply]
I read this article, but it didn't answer my questions. My puzzle is why the bottom part of the waveform diagram needs to be represented by the data of the right channel instead of the minimum negative value of the left channel

**Reply by katspaugh** [reply]
Because most audio files, especially music tracks, are in stereo, and you hear the two channels separately, not as a sum of their volumes.

**Reply by Idletalk** [reply]
Got it. Thank you

**Comment by jxu** [comment]
Is this documented on the site? The only place I've seen it mentioned is the source code.
It's not intuitive because you expect it to be plotting a raw signal like an oscilloscope or in Audacity, assuming one channel or mono if you remember most music is stereo.
I noticed the discrepancy when looking at the Beatles recording It Won't Be Long which uses hard-pan slowly moving around, and a section at the end where it's only the right channel.  The asymmetry looks like a weird DC offset when it's not.

**Reply by katspaugh** [reply]
I believe it's not explicitly documented anywhere, which I agree is not good. How would you prefer the waveform to look in case of stereo audio?

**Reply by jxu** [reply]
I think the current approach is fine, just that it should be documented, and ensure people know about the splitChannels option because I suspect that's what people will also be looking for.

**Reply by katspaugh** [reply]
Documented it in the FAQ, thanks for bringing this up!
